Toolkit/ArcGISToolkit/AR/ArcGISARView.swift

```diff
@@ -0,0 +1,527 @@
+    
+    /// Stop updating location and heading.
+    fileprivate func stopUpdatingLocationAndHeading() {
+        locationManager.stopUpdatingLocation()
+        if CLLocationManager.headingAvailable() {
+            locationManager.stopUpdatingHeading()
+        }
+    }
+
+    /// Start the locationManager with denied access.
+    fileprivate func startWithAccessDenied() {
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Start the locationManager with authorized access.
+    fileprivate func startWithAccessAuthorized() {
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Potential notification to the user of an error starting device tracking.
+    ///
+    /// - Parameter error: error that ocurred when starting tracking.
+    fileprivate func didStartOrFailWithError(_ error: Error?) {
+        if !notifiedStartOrFailure, let error = error {
+            // TODO: present error to user...
+            print("didStartOrFailWithError: \(String(reflecting:error))")
+        }
+        
+        notifiedStartOrFailure = true;
+    }
+    
+    /// Handle a change in authorization status to "denied".
+    fileprivate func handleAuthStatusChangedAccessDenied() {
+        // auth status changed to denied
+        stopUpdatingLocationAndHeading()
+        
+        // We were waiting for user prompt to come back, so notify.
+        didStartOrFailWithError(ArcGISARView.accessDeniedError())
+    }
+    
+    /// Handle a change in authorization status to "authorized".
+    fileprivate func handleAuthStatusChangedAccessAuthorized() {
+        // Auth status changed to authorized; now that we have authorization - start updating location and heading.
+        didStartOrFailWithError(nil)
+        startUpdatingLocationAndHeading()
+    }
+    
+    /// Called when device orientation changes.
+    ///
+    /// - Parameter notification: the notification.
+    @objc func orientationChanged(notification: Notification?) {
+        // Handle rotation here.
+        switch UIApplication.shared.statusBarOrientation {
+        case .landscapeLeft:
+            orientationQuat = simd_quaternion(0, 0, 1.0, 0);
+        case .landscapeRight:
+            orientationQuat = simd_quaternion(0, 0, 0, 1.0);
+        case .portrait:
+            orientationQuat = simd_quaternion(0, 0, sqrt(0.5), sqrt(0.5));
+        case .portraitUpsideDown:
+            orientationQuat = simd_quaternion(0, 0, -sqrt(0.5), sqrt(0.5));
+        default:
+            break
+        }
+    }
+    
+    // MARK: Errors
+    
+    /// Error used when `ARKit` is not supported on the current device.
+    ///
+    /// - Returns: Error stating `ARKit` not supported.
+    fileprivate class func notSupportedError() -> NSError {
+        let userInfo = [NSLocalizedDescriptionKey : "The device does not support ARKit functionality."]
+        return NSError(domain: AGSErrorDomain, code: ARError.unsupportedConfiguration.rawValue, userInfo: userInfo)
```

Comment:
I missed this one. If you're going to use an `ARError` (which is fine), it should be in that domain:
```suggestion
        return NSError(domain: ARErrorDomain, code: ARError.unsupportedConfiguration.rawValue, userInfo: userInfo)
```